packages/core/src/lib/select/select.svelte

```diff
@@ -25,9 +34,35 @@ export let value: string | undefined = undefined;
 export let disabled = false;
 
 /** The state of the select (info, warn, error, success), if any. */
-export let state: InputState = 'none';
+export let state: SelectState = 'none';
+
+const onInput = (event: Event) => {
+  if (disabled) {
+    event.preventDefault();
+    return;
+  }
+
+  dispatch('input', event);
+};
+
+const onMouseDown = (event: MouseEvent) => {
+  if (disabled) {
+    event.preventDefault();
+    return;
+  }
+
+  dispatch('mousedown', event);
+};
+
+const onKeyDown = (event: KeyboardEvent) => {
```

Comment:
What are these `...Down` events used for? They'll differ from a `click` or a key press, in that both of those won't fire until the mouse/key has been released (`...Up`)

Reply:
These are here to allow disabled selects to continue to be focusable/discoverable while behaving disabled. If we use the disabled attribute the select will not be discoverable, but selects don't have a readonly attribute like inputs. So what we want to do is capture events when the select is disabled and prevent them from doing anything if so. In the case of the `keydown` event we still want to allow the `tab` key to behave natively since we want to be able to tab away from disabled inputs. the `mousedown` one is to make sure you can't open the select menu. Otherwise if the select isn't disable we just dispatch the native event as usual.